airgun/entities/redhat_repository.py

```diff
@@ -17,13 +17,22 @@ def search(self, value, category='Available', types=None):
         view = self.navigate_to(self, 'All')
         return view.search(value, category=category, types=types)
 
-    def read(self, entity_name, category='Available', widget_names=None):
+    def read(self, entity_name=None, category='Available', recommended_repo=None):
         """Read RH Repositories values.
 
         :param entity_name: The repository name
         :param category: The repository category to search, options: Available, Enabled
+        :param recommended_repo: on/off RH recommended repositories
         """
         view = self.navigate_to(self, 'All')
+
+        if recommended_repo:
+            current_value = self.browser.get_attribute('class',
+                                                       locator=view.recommended_repos.locator)
+            if recommended_repo not in current_value:
+                view.recommended_repos.click()
+            return view.available.read()
```

Comment:
@vijay8451 Sorry, I have one more question. 

Do we want to make sure that the switcher should be always in OFF state? Just to ensure that if someone calls read method with the only entity_name it should give the correct result. (there could be chances where recommended_repo filter got applied in the previous test case)  
e.g. Scenarios1 :

- Test1 -> uses ->.read(recommended_repo='on')  => Result in recommended switch On
- Test2->uses ->.read(entity_name='$repo_name not exist in recommended repo') => Result not showing that repo.

I would recommend making recommended switcher always to Off State, anyone wants to result from ON state they can get by applying or we read with some logic if the search occurs with entity name it should check in off state as well.

Reply:
@omkarkhatavkar good question .. the switch always remain in `off` state and whenever a test enable that then 
- It keeps enable until page/browser refresh or session close.
- It will only keep on for that specific test not others.

Reply:
cool then everything works in that case.